src/Symfony/Component/HttpKernel/Controller/ArgumentResolverManager.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentResolver\ArgumentResolverInterface;
+
+/**
+ * The ArgumentResolverManager chains over the registered argument resolvers to
+ * resolve all controller arguments.
+ *
+ * @author Wouter J <wouter@wouterj.nl>
+ */
+class ArgumentResolverManager
+{
+    /**
+     * @var ArgumentResolverInterface[]
+     */
+    private $resolvers = array();
+
+    /**
+     * Adds an argument resolver.
+     *
+     * @param ArgumentResolverInterface $resolver
+     */
+    public function add(ArgumentResolverInterface $resolver)
+    {
+        $this->resolvers[] = $resolver;
+    }
+
+    public function getArguments(Request $request, $controller)
```

Comment:
yes. This class is designed to work with the ControllerResolver purely. The ControllerResolver already converts `Foo::bar` to `array($obj, $method)`, so there is no need to check for the `Foo::bar` syntax here again.